app/src/main/java/com/hedvig/app/feature/keygear/data/KeyGearItemsRepository.kt

```diff
@@ -106,4 +110,64 @@ class KeyGearItemsRepository(
                 files
             )
         ).toDeferred()
+
+    suspend fun uploadReceipt(itemId: String, file: Uri) {
+        val mimeType = fileService.getMimeType(file)
+        val uploadFile = File(
+            context.cacheDir,
+            fileService.getFileName(file)
+                ?: "${UUID.randomUUID()}.${fileService.getFileExtension(file.toString())}"
+        )
+        context.contentResolver.openInputStream(file)?.into(uploadFile)
+        val uploadResult = apolloClientWrapper
+            .apolloClient
+            .mutate(UploadFileMutation(FileUpload(mimeType, uploadFile)))
+            .toDeferred()
+            .await()
+
+        val uploadData = uploadResult.data()
+        if (uploadData == null) {
+            e { "Failed to upload photo" }
```

Comment:
Not sure if I like it, really not a fan of code that doesn't explain what it does.

Reply:
I'm not sure that I follow 🤔